swan-lake/resources/featured-scenarios/write-a-restful-api-with-ballerina.md

```diff
@@ -127,103 +187,75 @@ service /covid/status on new http:Listener(9000) {
     } else {
-        covidEntries.forEach(covdiEntry => covidTable.add(covdiEntry));
+        covidEntries.forEach(covidEntry => covidTable.add(covidEntry));
         return covidEntries;
     }
 }
 ```
 
 In this code:
 
-- It is chosen to either accept the entire payload or send back an error. Copying this straightway results in an error, which is expected as the `ConflictingIsoCodesError` type is not defined yet.
-- This resource has a resource argument named `covidEntries` annotated with `@http:Payload`. This means the resource is expecting a payload with the `CovideEntry[]` type. There are two types of records `CovideEntry[]` and `ConflictingIsoCodesError` that will be used as the return values.
-
-#### Define the error records
-
-To define the error records, add the code below to the API template file (i.e., `service.bal`).
-
-```ballerina
-public type ConflictingIsoCodesError record {|
-    *http:Conflict;
-    ErrorMsg body;
-|};
-
-public type ErrorMsg record {|
-    string errmsg;
-|};
-```
-
-In this code:
-- Ballerina uses `*http:Conflict` to denote that one type is a subtype of another. In this case, `ConflictingIsoCodesError` is a subtype of `http:Conflict`.
-- The body of the response is of type `ErrorMsg`, which simply has a string field named `errmsg`. Based on the need, users can have any data type for their response body.
-- Ballerina has a defined set of types for each HTTP status code. This allows you to write services in a type-oriented way, which in turn is helpful when it comes to tooling and generating OpenAPI specifications for HTTP services. 
+- It is chosen to accept the entire payload or send back an error.
+- This resource has an argument named `covidEntries`, which means the resource expects a payload with the `CovidEntry[]` type. Two types of records, `CovidEntry[]` and `ConflictingIsoCodesError`, will be used as the return types.
 
 ## Implement the second endpoint
 
 The second endpoint has only one resource to get COVID-19 data filtered by the ISO code.
 
 ### Create the resource of the second endpoint
 
-To create the resource of the second endpoin, add the code below to the API template file (i.e., `service.bal`).
+Similar to how you created the [second resource of the first endpoint](#create-the-second-resource-to-add-data), create the resource of the second endpoint below using the diagram view in VS Code.
+
+>**Note:** Define an HTTP resource that allows the `GET` operation on the resource path `countries` and accepts the `iso_code` path parameter. Define and use `InvalidIsoCodeError` along with `CovidEntry[]` as the response types.
+![Create second GET resource](/learn/images/featured-scenarios/write-a-restful-api-with-ballerina/create-second-get-resource.gif)
+
+Implement the logic of this `GET` resource function with the code below.
 
 ```ballerina
 resource function get countries/[string iso_code]() returns CovidEntry|InvalidIsoCodeError {
     CovidEntry? covidEntry = covidTable[iso_code];
```

Comment:
We can use the `hasKey` and `getKey` methods here right?

Reply:
Shall we do these code improvements separately? Kinda out of scope for this task, IMO.

Reply:
But, we'll have to redo the GIFs then, so may be we'll have to do that first.

Reply:
Shall we finalize the code in a comment before proceeding with the GIFs?

Reply:
Done.